### ISSUE TITLE
Voxforge dataset importer

### ISSUE THREAD

**Issue by gardenia22** [mozilla/DeepSpeech#714]
I wrote an importer for voxforge dataset(~100 hour), and use 1/100 data for test, 1/100 data for dev, following is the code:

```python
#!/usr/bin/env python
import sys
from glob import glob
from os import makedirs, path
import urllib2
import tarfile
import pandas
from BeautifulSoup import BeautifulSoup
from tensorflow.python.platform import gfile
from tensorflow.contrib.learn.python.learn.datasets import base

def _download_and_preprocess_data(data_dir):
    # Conditionally download data to data_dir
    if not path.isdir(data_dir):
        makedirs(data_dir)
    archive_dir = data_dir+"/archive"
    if not path.isdir(archive_dir):
        makedirs(archive_dir)
    print("Downloading Voxforge data set into {} if not already present...".format(archive_dir))
    voxforge_url = 'http://www.repository.voxforge1.org/downloads/SpeechCorpus/Trunk/Audio/Main/16kHz_16bit'
    html_page = urllib2.urlopen(voxforge_url)
    soup = BeautifulSoup(html_page)
    # list all links
    links = soup.findAll('a')
    refs = [l['href'] for l in links if ".tgz" in l['href']]
    def filename_of(x): return path.split(x)[1]

    for i, ref in enumerate(refs):
        print('Downloading {} / {} files'.format(i, len(refs)))
        download_url = voxforge_url + '/' + ref
        base.maybe_download(filename_of(download_url), archive_dir, download_url)

    # Conditionally extract data to dataset_dir
    print("Extracting Voxforge data set into {} if not already present...".format(data_dir))
    tarfiles = glob(path.join(archive_dir, "*.tgz"))
    if not path.isdir(path.join(data_dir,"test")):
        makedirs(path.join(data_dir,"test"))
    if not path.isdir(path.join(data_dir,"dev")):
        makedirs(path.join(data_dir,"dev"))
    if not path.isdir(path.join(data_dir,"train")):
        makedirs(path.join(data_dir,"train"))

    number_of_files = len(tarfiles)
    number_of_test = number_of_files/100
    number_of_dev = number_of_files/100
    for i,archive in enumerate(tarfiles):
        if i<number_of_test:
            dataset_dir = path.join(data_dir,"test")
        elif i<number_of_test+number_of_dev:
            dataset_dir = path.join(data_dir,"dev")
        else:
            dataset_dir = path.join(data_dir,"train")
        if not gfile.Exists(path.join(dataset_dir, '.'.join(filename_of(archive).split(".")[:-1]))):
            tar = tarfile.open(archive)
            tar.extractall(dataset_dir)
            tar.close()
            print(('Extracting {} / {} files'.format(i, len(tarfiles))))
            print archive

    # Generate data set
    print("Generating Voxforge data set into {}".format(data_dir))
    test_files = _generate_dataset(data_dir, "test")
    dev_files = _generate_dataset(data_dir, "dev")
    train_files = _generate_dataset(data_dir, "train")

    # Write sets to disk as CSV files
    train_files.to_csv(path.join(data_dir, "voxforge-train.csv"), index=False)
    dev_files.to_csv(path.join(data_dir, "voxforge-dev.csv"), index=False)
    test_files.to_csv(path.join(data_dir, "voxforge-test.csv"), index=False)


def _generate_dataset(data_dir, data_set):
    extracted_dir = path.join(data_dir, data_set)
    files = []
    for promts_file in glob(path.join(extracted_dir+"/*/etc/", "PROMPTS")):
        if path.isdir(path.join(promts_file[:-11],"wav")):
            with open(promts_file) as f:
                for line in f:
                    id = line.split(' ')[0].split('/')[-1]
                    transcript = line.split(' ')[1:]
                    transcript = ' '.join(transcript).replace('\n', '').lower()
                    wav_file = path.join(promts_file[:-11],"wav/" + id + ".wav")
                    if gfile.Exists(wav_file):
                        wav_filesize = path.getsize(wav_file)
                        # remove audios that are shorter than 0.5s, longer than 20s.
                        # remove audios that are too short for transcript.
                        if (wav_filesize/32000)>0.5 and (wav_filesize/32000)<20 and transcript!="" and \
                            wav_filesize/len(transcript)>1400:
                            files.append((path.abspath(wav_file), wav_filesize, transcript))

    return pandas.DataFrame(data=files, columns=["wav_filename", "wav_filesize", "transcript"])

if __name__=="__main__":
    _download_and_preprocess_data(sys.argv[1])
```

**Comment by kdavis-mozilla** [comment]
Awesome @gardenia22 !

We'd love to have your Voxforge importer contribution. Would it be possible to open a pull request so we can integrate it into our code base? 

The details of opening a pull request using github are give here[[1](https://help.github.com/articles/about-pull-requests/)][[2](https://help.github.com/articles/creating-a-pull-request/)].

Thanks again!

**Comment by gardenia22** [comment]
@kdavis-mozilla created a PR #716

**Comment by kdavis-mozilla** [comment]
@gardenia22 Thanks!

**Comment by kdavis-mozilla** [comment]
Closed with PR #716
